app/services/complete_moab_service/check_existence.rb

```diff
@@ -60,5 +33,38 @@ def update_complete_moab_preserved_object_or_set_status
         moab_validator.update_status('invalid_moab')
       end
     end
+
+    def check_versions
+      Rails.logger.debug "check_existence #{druid} called"
+
+      with_active_record_transaction_and_rescue do
+        raise_rollback_if_version_mismatch
+
+        return report_results! unless moab_validator.can_validate_current_comp_moab_status?
+
+        if incoming_version == complete_moab.version
+          moab_validator.set_status_as_seen_on_disk(true) unless complete_moab.status == 'ok'
+          results.add_result(AuditResults::VERSION_MATCHES, 'CompleteMoab')
+        elsif incoming_version > complete_moab.version
+          moab_validator.set_status_as_seen_on_disk(true) unless complete_moab.status == 'ok'
+          results.add_result(AuditResults::ACTUAL_VERS_GT_DB_OBJ, db_obj_name: 'CompleteMoab', db_obj_version: complete_moab.version)
+          update_complete_moab_preserved_object_or_set_status
+        else # incoming_version < complete_moab.version
+          moab_validator.set_status_as_seen_on_disk(false)
+          results.add_result(AuditResults::ACTUAL_VERS_LT_DB_OBJ, db_obj_name: 'CompleteMoab', db_obj_version: complete_moab.version)
+        end
+        complete_moab.update_audit_timestamps(moab_validator.ran_moab_validation?, true)
+        complete_moab.save!
+      end
+    end
+
+    def report_missing
```

Comment:
I noticed that three of the other subclasses have similar logic around adding AuditResults? Maybe this is another candidate for refactoring as part of another task?

Reply:
You're right. Let me get that.